### DISCUSSION TITLE
Allow overriding autoload order via override.cfg (to make modding for Godot games better)

### DISCUSSION THREAD

**Discussion by Qubus0** [godotengine/godot-proposals#6137] [General Discussions]
As mentioned in [this issue #71589 in the main repo](https://github.com/godotengine/godot/issues/71589#issue-1537236169)

Recap:
Overriding autoloads with an override.cfg ignores the order set in the override. Newly defined autoloads are always loaded last.

Why does this matter? I'm trying to (co-)create a [mod loader](https://github.com/GodotModding/godot-mod-loader) that works with any Godot game. The loader itself is an autoload. Using the override.cfg, we can 'install' it into a game without having to touch or modify its source code. The loader always being last limits the capabilities for extending existing autoloads heavily. 

The details of the issue were very well explained by @akien-mga, including an idea
> `override.cfg` does override the _values_ of existing _properties_, and appends new property/value pairs which aren't present in the original `project.godot`.
> 
> The problem here is that autoloads as registered as _properties_ themselves. And there doesn't seem to be a mechanism for removing existing properties.
> 
> Changing it to e.g. a dictionary property could make the dictionary overridable, e.g.:
> 
> ```
> [autoload]
> 
> autoloads={
>   "AutoloadOverride": "*res://autoload_override.gd",
>   "Autoload1": "*res://autoload_1.gd",
>   "Autoload2": "*res://autoload_2.gd",
> }
> ```
> 
> But that's indeed material for a feature proposal, so we can discuss possible solutions and implement the best one.

A change like this will probably be the easiest to implement, but other ways of "inserting" an autoload (or any script that can run before everything starts) into an existing game would be  greatly appreciated.

Let me know what you think

**Comment by Qubus0** [comment]
If we can come to a conclusion on how to implement it, I am completely willing to give it a try myself, even though i've never done something like this before - but that's just experience waiting to be gathered, right? ;)

**Comment by Ryhon0** [comment]
Assigning an autoload to null and assigning it again allows you to change it's order, e.g.
```tres
[autoload]
Auto1=null
MyAuto="*res://myauto.gd"
Auto1="*res://auto1.gd"
```
This won't work if you're using a ConfigFile, though because setting to null removes the key instead of creating a null entry.
You can't add duplicate keys either.
I worked around this by defining special sequences, which will be replaced after serializing to a string

**Reply by Qubus0** [reply]
nice tip, thank you. we'll give it a go 😄

**Comment by bruvzg** [comment]
This is already possible by setting existing autoloads to `null` first:

```ini
[autoload]

;Unset existing autoloads
Autoload1=null
Autoload2=null

;Set new autoloads
AutoloadOverride="*res://autoload_override.gd"
Autoload1="*res://autoload_1.gd"
;Autoload2="*res://autoload_2.gd"
```

result in:
```
autoload override init
autoload 1 init
autoload override ready
autoload 1 ready
```

**Comment by Qubus0** [comment]
closed by https://github.com/godotengine/godot/pull/113078
